ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/filter/FilterManager.java

```diff
@@ -280,6 +262,23 @@ void recordPendingTransactionEvent(final Transaction transaction) {
     final long toBlockNumber =
         filter.getToBlock().getNumber().orElse(blockchainQueries.headBlockNumber());
 
-    return blockchainQueries.matchingLogs(fromBlockNumber, toBlockNumber, filter.getLogsQuery());
+    return findLogsWithinRange(filter, fromBlockNumber, toBlockNumber);
+  }
+
+  private List<LogWithMetadata> findLogsWithinRange(
```

Comment:
This method was introduced to avoid duplications. It is used by two methods in this class.